src/flux/internal.c

```diff
@@ -0,0 +1,187 @@
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ ***************************************************************************/
+
+
+#include <stdio.h>
+#include <jansson.h>
+#include <time.h>
+#include <limits.h>
+#include <flux/core.h>
+
+#include "internal_yogrt.h"
+
+#define BOGUS_TIME -1
+
+int jobid_valid = 0;
+int verbosity = 0;
+
+static flux_jobid_t jobid = 0;
+
+int internal_init(int verb)
+{
+	char *jobid_str;
+
+	verbosity = verb;
+	jobid_valid = 0;
+
+	if ((jobid_str = getenv("FLUX_JOB_ID")) == NULL) {
+		error("ERROR: FLUX_JOB_ID is not set."
+			  " Remaining time will be a bogus value.\n");
+		return jobid_valid;
+	}
+
+	if (flux_job_id_parse(jobid_str, &jobid) < 0) {
+		error("ERROR: Unable to parse FLUX_JOB_ID %s."
+			  " Remaining time will be a bogus value.\n", jobid_str);
+		return jobid_valid;
+	}
+
+	jobid_valid = 1;
+
+	return jobid_valid;
+}
+
+char *internal_backend_name(void)
+{
+	return "FLUX";
+}
+
+static int get_job_expiration(flux_jobid_t id, long int *expiration)
+{
+	flux_t *h = NULL;
+	flux_t *child_handle = NULL;
+	flux_future_t *f;
+	json_t *job;
+	json_t *value;
+	double exp;
+	const char *uri = NULL;
+	int rc = -1;
+
+	if (!(h = flux_open(NULL, 0))) {
+		error("ERROR: flux_open() failed with errno %d\n", errno);
+		goto out;
+	}
+
+	/*
+	 * Determine whether to ask our parent or not
+	 * See https://github.com/flux-framework/flux-core/issues/3817
+	 */
+
+	if (!getenv("FLUX_KVS_NAMESPACE")) {
+		uri = flux_attr_get(h, "parent-uri");
+		if (!uri) {
+			error("ERROR: no FLUX_KVS_NAMESPACE and flux_attr_get failed with "
+				  "errno %d\n", errno);
+			goto out;
+		}
+
+		child_handle = h;
+		h = flux_open(uri, 0);
+		if (!h) {
+			printf("flux_open with parent-uri %s failed with errno %d\n", uri,
+				   errno);
+			goto out;
+		}
+	}
+
+	if (!(f = flux_job_list_id(h, jobid, "[\"expiration\"]"))) {
+		error("ERROR: flux_job_list failed with errno %d.\n", errno);
+		goto out;
+	}
+
+	if (flux_rpc_get_unpack(f, "{s:o}", "job", &job) < 0) {
+		error("ERROR: flux_rpc_get_unpack failed with errno %d.\n", errno);
+		goto out;
+	}
+
+	if (!(value = json_object_get(job, "expiration"))) {
+		error("ERROR: flux_object_get for id failed.\n");
+		goto out;
+	}
+
+	if ((exp = json_real_value(value)) == 0.0) {
+		error("ERROR: json_real_value failed.\n");
+		goto out;
+	}
+
+	*expiration = (long int) exp;
+	rc = 0;
+
+out:
+
+	if (f) {
+		flux_future_destroy(f);
+	} if (h) {
+		flux_close(h);
+	}
```

Comment:
No need to check for `NULL` before `flux_future_destroy()` and `flux_close()`. Both of these functions are no-ops with `NULL` args.

Reply:
Great, thanks.